File: PHP/MinimumLimitOfBallsInaBagTest.php
<?php
/**
 * @version PHP 8.2.22
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit MinimumLimitOfBallsInaBagTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MinimumLimitOfBallsInaBag.php");

class MinimumLimitOfBallsInaBagTest extends TestCase {

    public function testMinimumLimitOfBallsInaBag() {
        $obj = new MinimumLimitOfBallsInaBag();
        //Test 1
        $want = 3;
        $got = $obj->minimumSize( [9], 2 );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 2;
        $got = $obj->minimumSize( [2,4,8,2], 4 );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/MinimumLimitOfBallsInaBag_test.go
/**
 * @version GO 1.22.3
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestMinimumLimitOfBallsInaBag
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMinimumLimitOfBallsInaBag(t *testing.T) {
	tables := []struct {
		want int
		nums []int
		max  int
	}{
		{3, []int{9}, 2},
		{2, []int{2, 4, 8, 2}, 4},
	}

	for _, table := range tables {
		got := minimumSize(table.nums, table.max)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
